hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataLogRecordReader.java

```diff
@@ -150,9 +151,9 @@ private static Predicate<String> createPrefixMatchingPredicate(List<String> keyP
    * Builder used to build {@code HoodieMetadataMergedLogRecordScanner}.
    */
   public static class Builder {
-    private final HoodieMergedLogRecordScanner.Builder scannerBuilder =
-        new HoodieMergedLogRecordScanner.Builder()
-            .withKeyFiledOverride(HoodieMetadataPayload.KEY_FIELD_NAME)
+    private final HoodieMetadataMergedLogRecordScanner.Builder scannerBuilder =
+        new HoodieMetadataMergedLogRecordScanner.Builder()
```

Comment:
Do we need the scanner for all kinds of metadata index types? The full outer join may bring in perforamce regression.

Reply:
No, we need only for secondary index. I have changes in a subsequent PR. But, i'll do as suggested in this PR itself.